tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/MicroSecEndTest.java

```diff
@@ -0,0 +1,468 @@
+                .build();
+    }
+
+    public void performAnalysis(String model, int variant, Set<Integer> violationsSet) {
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+
+        assertFalse(flowGraph.getTransposeFlowGraphs()
+                .isEmpty());
+        Map<Integer, List<AbstractTransposeFlowGraph>> violatingTransposeFlowGraphs = new HashMap<>();
+        for (var transposeFlowGraph : flowGraph.getTransposeFlowGraphs()) {
+
+            hasLoggingServer(transposeFlowGraph, violatingTransposeFlowGraphs);
+
+            hasSecretManager(transposeFlowGraph, violatingTransposeFlowGraphs);
+
+            for (var vertex : transposeFlowGraph.getVertices()) {
+                hasGateway(violationsSet, variant, vertex);
+
+                hasAuthenticatedRerquest(violationsSet, variant, vertex);
+
+                hasOptionalAuthorizedEntrypoint(violationsSet, variant, vertex, flowGraph.getTransposeFlowGraphs());
+
+                hasTransformedEntryIdentity(violationsSet, variant, vertex);
+
+                hasTokenValidation(violationsSet, variant, vertex);
+
+                hasLoginAttemptsRegulation(violationsSet, variant, vertex);
+
+                hasEncryptedEntryConnection(violationsSet, variant, vertex);
+
+                hasEncrytedInternalConnection(violationsSet, variant, vertex);
+
+                hasLocalLogging(violationsSet, variant, vertex);
+
+                hasLogSanitization(violationsSet, variant, vertex);
+
+                hasOptionalMessageBroker(violationsSet, variant, vertex, flowGraph.getTransposeFlowGraphs());
+
+            }
+        }
+        checkCrossTransposeFlowGraphViolations(flowGraph, violatingTransposeFlowGraphs, violationsSet);
+
+    }
+
+    @Test
+    void testConstraints() {
+        for (var model : TUHH_MODELS.keySet()) {
+            for (int variant : TUHH_MODELS.get(model)) {
+                Set<Integer> violationSet = new TreeSet<Integer>();
+                String variationName = model + "_" + variant;
+                performAnalysis(Paths.get(location, model, variationName)
+                        .toString(), variant, violationSet);
+                logger.info("Variant: " + variationName);
+                logger.info("Violations: " + violationSet);
+                assertFalse(violationSet.contains(variant));
+            }
+        }
+    }
+
+    @Test
+    void testDetectCycles() {
+        var model = Paths.get(location, "anilallewar", "anilallewar_0")
+                .toString();
+
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+
+        assertTrue(flowGraph.wasCyclic());
+
+    }
+    
+    
+    @Test
+    void testRamCopyIssue() {
+        var model = Paths.get(location, "sqshq", "sqshq_18")
```

Comment:
How long does it take to check sqshk_18 50 times? Just to make sure; it does contain loops, right?

Reply:
about 4 secends each so 3.5min

Reply:
Would running the test 10 times suffice?
40 seconds wouldn't be that bad on an increase.

Reply:
i feel like 15 should be enough

Reply:
alright